src/test/run-make-fulldeps/extern-flag-fun/Makefile

```diff
@@ -4,7 +4,6 @@ all:
 	$(RUSTC) bar.rs --crate-type=rlib
```

Comment:
Is this the only test?

Reply:
`extern-flag-fun` is not actually testing this feature. It is a 5-year old test from when `--extern` was first added. The removed `--extern hello` line was originally there to verify that it would error due to the missing path. When pathless `--extern` was added, the error changed to "missing -Z unstable-options". With this PR, the `hello` line no longer fails (and thus breaks the test), due to `-L` flags injected by compiletest. I figured it no longer fit the spirit of the test.

I have added some tests.